src/plugins/intel_cpu/src/transformations/snippets/x64/pass/brgemm_to_brgemm_cpu.cpp

```diff
@@ -58,24 +58,22 @@ pass::BrgemmToBrgemmCPU::BrgemmToBrgemmCPU() {
         if (!brgemm || brgemm_plugin)
             OPENVINO_THROW("BrgemmCPU cannot be in body before BrgemmToBrgemmCPU pass");
 
-        if (brgemm->is_dynamic()) {
-            return false;
-        }
-
         const auto& brgemm_in0_desc = PortDescriptorUtils::get_port_descriptor_ptr(brgemm->input(0));
         const auto& brgemm_in1_desc = PortDescriptorUtils::get_port_descriptor_ptr(brgemm->input(1));
         const auto& brgemm_out_desc = PortDescriptorUtils::get_port_descriptor_ptr(brgemm->output(0));
 
-        const auto dimsMatMulIn0 = snippets::utils::get_planar_pshape(brgemm->input(0)).get_shape();
-        const auto dimsMatMulIn1 = snippets::utils::get_planar_pshape(brgemm->input(1)).get_shape();
+        const auto dimsMatMulIn0 = snippets::utils::get_planar_pshape(brgemm->input(0));
+        const auto dimsMatMulIn1 = snippets::utils::get_planar_pshape(brgemm->input(1));
 
         const auto K = *dimsMatMulIn0.rbegin();
         const auto N = *dimsMatMulIn1.rbegin();
 
         const auto element_type_a = brgemm->get_input_element_type(0);
         const auto brgemmVNNIFactor = 4 / element_type_a.size();
         const bool isAMXSupported = dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx);
-        const bool with_amx = isAMXSupported && element_type_a != ov::element::f32 && (K % brgemmVNNIFactor == 0) && (N % brgemmVNNIFactor == 0);
+        const bool with_amx = isAMXSupported && element_type_a != ov::element::f32 &&
+                              K.is_static() && K.get_length() % brgemmVNNIFactor == 0 &&
+                              N.is_static() && N.get_length() % brgemmVNNIFactor == 0;
```

Comment:
That's right, but the callback handles common limitation, and here the check is relevant to a cpu node.
I mean, that if we'll support another (say TPP) matmul, then this check here will be used. Consider TPP tests for example, when the callback is ignored.

> BTW do you know whether matmul CPU implementation uses amx implementation in case of dynamic KN?
Dynamism is supported for f32 only, but since the palette is actually a [part of the compiled kernel](https://github.com/IvanNovoselov/openvino/blob/e424c0fc65d198040363618e76a5b5a3288b1797/src/plugins/intel_cpu/src/emitters/snippets/x64/kernel_executors/brgemm.hpp#L64-L69), and the kernel is recompiled for every shape, there shouldn't be any problems with supporting it.